Cython/Compiler/Nodes.py

```diff
@@ -2363,7 +2363,9 @@ def align_error_path_gil_to_success_path(): pass
             if entry.type.needs_refcounting:
                 if entry.is_arg and not entry.cf_is_reassigned:
                     continue
-                assure_gil('success')
+                if not entry.type.is_memoryviewslice:
```

Comment:
I was wondering if we should add a `refcounting_needs_gil` field to the type to try to help with future extensibility rather than just special casing on memoryviewslice.

With that said, looks good

Reply:
Done please have a look.